invenio_swh/tasks.py

```diff
@@ -83,16 +85,49 @@ def poll_deposit(self, id_):
     try:
         deposit = service.read(id_).deposit
         service.sync_status(deposit.id)
-        new_status = deposit.status
-    except DepositFailed:
-        # If the deposit already failed, we don't need to retry.
+    except Exception:
+        # Gracefully fail, the deposit can still be retried
+        pass
+
+    # If the deposit failed already, don't do anything else
+    if deposit.status == SWHDepositStatus.FAILED:
         return
 
     # Manually set status to FAILED on last retry.
     # Celery has a bug where it doesn't raise MaxRetriesExceededError, therefore we need to check retries manually.
     if self.request.retries == 5:
-        service.handle_status_update(deposit, SWHDepositStatus.FAILED)
+        service.update_status(deposit, SWHDepositStatus.FAILED)
         return
 
-    if new_status == SWHDepositStatus.WAITING:
+    if deposit.status == SWHDepositStatus.WAITING:
         raise DepositWaiting("Deposit is still waiting")
+
+
+@shared_task()
+def cleanup_depositions():
+    """Cleanup old depositions."""
+    if not current_app.config.get("SWH_ENABLED"):
+        current_app.logger.warning(
+            "Sofware Heritage interation is not enabled, cleanup task can't run."
+        )
+        return
+    Deposit = service.record_cls.model_cls
+    # query for records that are stuck in "waiting"
+    res = Deposit.query.filter(
+        Deposit.status == SWHDepositStatus.WAITING,
+        Deposit.updated < datetime.now() - timedelta(days=1),
+    )
+
+    for deposit in res:
+        try:
+            service.sync_status(deposit.id)
+        except Exception as ex:
+            # If the sync failed for any reason, set the status to "FAILED"
+            try:
```

Comment:
ugly nested try / catch, but `service.update_status` can also fail. In that case, we want to continue trying other deposits